ray_julia_jll/src/wrappers/any.jl

```diff
@@ -167,8 +168,26 @@ FromNil(::Type{ObjectID}) = ObjectIDFromNil()
 
 ObjectID(str::AbstractString) = FromHex(ObjectID, str)
 
-Base.show(io::IO, x::ObjectID) = show(io, "ObjectID(\"$(Hex(x))\")")
-Base.:(==)(a::ObjectID, b::ObjectID) = Hex(a) == Hex(b)
+Base.show(io::IO, x::ObjectID) = write(io, "ObjectID(\"", Hex(x), "\")")
+
+# cannot believe I'm doing this...
+# 
+# Because ObjectID is a CxxWrap-defined type, it has two subtypes:
+# `ObjectIDAllocated` and `ObjectIDDereferenced`.  The first is returned when we
+# construct directly or return by value, the second when you pull a ref out of
+# say `std::vector<ObjectID>`.
+#
+# ObjectID is abstract, so the normal method definition:
+# 
+# Base.:(==)(a::ObjectID, b::ObjectID) = Hex(a) == Hex(b)
+#
+# is shadowed by more specific fallbacks defined by CxxWrap.
+for Ta in subtypes(ObjectID)
+    for Tb in subtypes(ObjectID)
+        @eval Base.:(==)(a::$Ta, b::$Tb) = Hex(a) == Hex(b)
+    end
+end
```

Comment:
Probably should do:
```suggestion
for Ta in subtypes(ObjectID), Tb in subtypes(ObjectID)
    @eval Base.:(==)(a::$Ta, b::$Tb) = Hex(a) == Hex(b)
end
```

If we want to drop the extra dependency I'd be good with doing:
```suggestion
for Ta in (ObjectIDAllocated, ObjectIDDereferenced), Tb in (ObjectIDAllocated, ObjectIDDereferenced)
    @eval Base.:(==)(a::$Ta, b::$Tb) = Hex(a) == Hex(b)
end
```
I feel like CxxWrap is clear about the naming of these subtypes so we don't need to do it dynamically. Also, if an abstract subtype is added this may no longer be specific enough again.